### PR TITLE
Added support for sorting and output formats on Minimap2 wrapper

### PR DESCRIPTION
**Pull request by fgvieira**
If I do not specify the `sort` option, I get an error:
```
Traceback (most recent call last):
  File "/lustre/hpc/hologenomics/fgvieira/NGS/snakemake-wrappers/bio/minimap2/aligner/test/.snakemake/scripts/tmpigon5oti.wrapper.py", line 57, in <module>
    raise ValueError("Unexpected value for params.sort ({})".format(sort))
ValueError: Unexpected value for params.sort (functools.partial(<function Namedlist._used_attribute at 0x7f8eb0aa7dc0>, _name='sort'))
```

It seems sort is being set as `functools.partial(<function Namedlist._used_attribute at 0x7f8eb0aa7dc0>, _name='sort')`, even though it should be set to `none` by default:
```
sort = snakemake.params.get("sort", "none")
```

Any idea why?

### CODE REVIEW

**Comment by dlaehnemann** [comment]
If sort is not specified at all, it should default to `"none"`. But are you referring to the case where you set `sort = ""` the empty string? I guess this would mean that sort is actually set to the empty string, so that the equality `if sort == "none"` doesn't hold. Could that be it? If so, maybe we should explicitly handle this (and default to "none" for the empty string, as well)?

**Comment by fgvieira** [comment]
I get that error if `params.sort` is not specified at all:
```
rule minimap2_bam:
    input:
        target="target/{input1}.mmi", # can be either genome index or genome fasta
        query=["query/reads1.fasta", "query/reads2.fasta"]
    output:
        "aligned/{input1}_aln.bam"
    log:
        "logs/minimap2/{input1}.log"
    params:
        extra="-x map-pb",           # optional
        sort_extra=""                # optional: extra arguments for samtools/picard
    threads: 3
    wrapper:
        "master/bio/minimap2/aligner"
```

It seems as if the default `none` is not being set. Can you double check on your side?

**Comment by jafors** [comment]
Seems like `params.get` is not a big fan of params named `sort`. If I rename it to `sorting`, everything seems to work. :smile:

**Comment by jafors** [comment]
I get the same errors. Renaming the parameter seems to fix that, however. `Sort` seems to be an unlucky choice of name.

**Comment by fgvieira** [comment]
Any suggestions?

File: bio/minimap2/aligner/test/Snakefile [hunk 1]
```diff
@@ -7,7 +7,41 @@ rule minimap2:
     log:
         "logs/minimap2/{input1}.log"
     params:
-        extra="-x map-pb"  # optional
+        extra="-x map-pb",           # optional
+        sort="coordinate",           # optional: Enable sorting. Possible values: 'none', 'queryname' or 'coordinate'
```
Comment (jafors) [inline bio/minimap2/aligner/test/Snakefile:11]:
```suggestion
        sorting="coordinate",           # optional: Enable sorting. Possible values: 'none', 'queryname' or 'coordinate'
```

File: bio/minimap2/aligner/test/Snakefile [hunk 2]
```diff
@@ -7,7 +7,41 @@ rule minimap2:
...
+    input:
+        target="target/{input1}.mmi", # can be either genome index or genome fasta
+        query=["query/reads1.fasta", "query/reads2.fasta"]
+    output:
+        "aligned/{input1}_aln.sam"
+    log:
+        "logs/minimap2/{input1}.log"
+    params:
+        extra="-x map-pb",           # optional
+        sort="none",                 # optional: Enable sorting. Possible values: 'none', 'queryname' or 'coordinate'
```
Comment (jafors) [inline bio/minimap2/aligner/test/Snakefile:27]:
```suggestion
        sorting="none",                 # optional: Enable sorting. Possible values: 'none', 'queryname' or 'coordinate'
```

File: bio/minimap2/aligner/test/Snakefile [hunk 3]
```diff
@@ -7,7 +7,41 @@ rule minimap2:
...
+    input:
+        target="target/{input1}.mmi", # can be either genome index or genome fasta
+        query=["query/reads1.fasta", "query/reads2.fasta"]
+    output:
+        "aligned/{input1}_aln.bam"
+    log:
+        "logs/minimap2/{input1}.log"
+    params:
+        extra="-x map-pb",           # optional
+        sort="coordinate",           # optional: Enable sorting. Possible values: 'none', 'queryname' or 'coordinate'
```
Comment (jafors) [inline bio/minimap2/aligner/test/Snakefile:43]:
```suggestion
        sorting="coordinate",           # optional: Enable sorting. Possible values: 'none', 'queryname' or 'coordinate'
```

File: bio/minimap2/aligner/wrapper.py [hunk 4]
```diff
@@ -3,14 +3,50 @@
...
 inputQuery = " ".join(snakemake.input.query)
 
+# Extract output format
+out_name, out_ext = path.splitext(snakemake.output[0])
+out_ext = out_ext[1:].upper()
+
+# Extract arguments.
+extra = snakemake.params.get("extra", "")
+
+sort = snakemake.params.get("sort", "none")
```
Comment (jafors) [inline bio/minimap2/aligner/wrapper.py:19]:
```suggestion
sort = snakemake.params.get("sorting", "none")
```

**Review by jafors — COMMENTED** [review]
This seems to work on my local machine.

**Review by jafors — APPROVED** [review]

**Pull request merged**